src/components/data/PathListAutomation.js

```diff
@@ -0,0 +1,405 @@
+    const { t: i18nCommon } = useTranslation("common");
+
+    let infoTypesCache = queryCache.getQueryData(INFO_TYPES_QUERY_KEY);
+
+    useQuery({
+        queryKey: INFO_TYPES_QUERY_KEY,
+        queryFn: getInfoTypes,
+        config: {
+            enabled: infoTypesQueryEnabled,
+            onSuccess: (resp) => {
+                setInfoTypeError(null);
+                setInfoTypes(resp.types);
+            },
+            staleTime: Infinity,
+            cacheTime: Infinity,
+            onError: (e) => {
+                setInfoTypeError(e);
+            },
+        },
+    });
+
+    const [createPathListFile, { status }] = useMutation(
+        ({ submission }) => pathListCreator(submission),
+        {
+            onSuccess: (data, { onSuccess }) => {
+                onSuccess();
+                setCreatePathListError(null);
+                onCreatePathList(data.file.id, data.file.path);
+            },
+            onError: (error, { onError }) => {
+                let errMsg = null;
+                const err_code = getErrorCode(error);
+                switch (err_code) {
+                    case ERROR_CODES.ERR_EXISTS:
+                        errMsg = t("pathListExistError");
+                        onError(err_code);
+                        break;
+
+                    case ERROR_CODES.ERR_NOT_FOUND:
+                        errMsg = t("pathListNoMatchError");
+                        onError();
+                        break;
+
+                    default:
+                        errMsg = t("infoTypeFetchError");
+                        onError();
+                }
+                setCreatePathListError(error);
+                setErrorMsg(errMsg);
+            },
+        }
+    );
+
+    useEffect(() => {
+        if (!infoTypesCache || infoTypesCache.length === 0) {
+            setInfoTypesQueryEnabled(true);
+        } else {
+            if (infoTypes === null || infoTypes.length === 0) {
+                setInfoTypes(infoTypesCache.types);
+            }
+        }
+    }, [infoTypes, infoTypesCache]);
+
+    const handleToggle = (value) => () => {
+        const currentIndex = selectedInfoTypes.indexOf(value);
+        const newChecked = [...selectedInfoTypes];
+
+        if (currentIndex === -1) {
+            newChecked.push(value);
+        } else {
+            newChecked.splice(currentIndex, 1);
+        }
+
+        setSelectedInfoTypes(newChecked);
+    };
+
+    const handlePathListCreation = (values, actions) => {
+        const { selectedPaths, pattern, dest, foldersOnly } = values;
+        actions.setSubmitting(true);
+
+        const submission = {
+            paths: selectedPaths,
+            dest,
+            pattern,
+            foldersOnly: foldersOnly || false,
+            recursive: true,
+            requestedInfoType,
+            selectedInfoTypes,
+        };
+        const onSuccess = () => {
+            actions.setSubmitting(false);
```

Comment:
🤔 It's been a while since I've worked with Formik, but I don't recall ever having to manually manage the submitting state.  I might be missing something though.